pkg/cloudprovider/servicecontroller/servicecontroller.go

```diff
@@ -445,13 +464,39 @@ func portsEqual(x, y *api.Service) bool {
 	if err != nil {
 		return false
 	}
-	if len(xPorts) != len(yPorts) {
+	return intSlicesEqual(xPorts, yPorts)
+}
+
+func intSlicesEqual(x, y []int) bool {
```

Comment:
This looks like a func that should be in a util package (or might be already). If you think it's reusable, I'd move it there so that we don't get similar functions sprinkled around the code base.